exporter/datadogexporter/metadata/host_test.go

```diff
@@ -106,6 +107,16 @@ func TestHostnameFromAttributes(t *testing.T) {
 	assert.True(t, ok)
 	assert.Equal(t, hostname, testHostName)
 
+	// Azure cloud provider means relying on the Azure function
+	attrs = testutils.NewAttributeMap(map[string]string{
+		conventions.AttributeCloudProvider: conventions.AttributeCloudProviderGCP,
```

Comment:
Shouldn't this be:
```suggestion
		conventions.AttributeCloudProvider: conventions.AttributeCloudProviderAzure,
```

otherwise you're testing the GCP case once again?